test/src/org/labkey/test/commands/mobileappstudy/ResolveEnrollmentTokenCommand.java

```diff
@@ -0,0 +1,123 @@
+ * limitations under the License.
+ */
+package org.labkey.test.commands.mobileappstudy;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.json.simple.JSONObject;
+import org.labkey.test.WebTestHelper;
+import org.labkey.test.data.mobileappstudy.ResolveEnrollmentTokenResponse;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+
+public class ResolveEnrollmentTokenCommand extends MobileAppCommand
+{
+    protected static final String CONTROLLER_NAME = "mobileappstudy";
+    protected static final String ACTION_NAME = "resolveenrollmenttoken";
+
+    private String _batchToken;
+    private String _projectName;
+
+    private String _studyId;
+    private String _message;
+
+    public String getProjectName()
+    {
+        return _projectName;
+    }
+    public void setProjectName(String projectName)
+    {
+        _projectName = projectName;
+    }
+
+    public ResolveEnrollmentTokenCommand(String project, String batchToken, Consumer<String> logger)
+    {
+        _batchToken = batchToken;
+        _projectName = project;
+
+        setLogger(logger);
+    }
+
+    public String getBatchToken()
+    {
+        return _batchToken;
+    }
+    public void setBatchToken(String batchToken)
+    {
+        _batchToken = batchToken;
+    }
+
+    public String getStudyId()
+    {
+        return _studyId;
+    }
+
+    public String getMessage()
+    {
+        return _message;
+    }
+
+    @Override
+    public HttpResponse execute(int expectedStatusCode)
+    {
+        HttpPost post = new HttpPost(getTargetURL());
+        return execute(post, expectedStatusCode);
+    }
+
+    @Override
+    public String getTargetURL()
+    {
+        Map<String, String> params = new HashMap<>();
+        if (StringUtils.isNotBlank(getBatchToken()))
+            params.put("token", getBatchToken());
+        return WebTestHelper.buildURL(CONTROLLER_NAME, getProjectName(), ACTION_NAME, params);
+    }
+
+    @Override
+    public String getBody()
+    {
+        return "";
+    }
+
+    @Override
+    protected void parseErrorResponse(JSONObject response)
+    {
+        super.parseErrorResponse(response);
+        _message = (String)response.get("message");
+    }
+
+    @Override
+    protected void parseSuccessfulResponse(JSONObject response)
+    {
+        ObjectMapper mapper = new ObjectMapper();
+        try
+        {
+            ResolveEnrollmentTokenResponse enrollmentTokenResponse = mapper.readValue(response.toJSONString(), ResolveEnrollmentTokenResponse.class);
+            if (null != enrollmentTokenResponse)
+                _studyId = enrollmentTokenResponse.getStudyId();
+        }
+        catch (IOException e)
+        {
+            //TODO: do something here...
```

Comment:
Just rethrow. No reason to swallow `IOException`. Something probably went way sideways.
```suggestion
            throw new RuntimeException(e);
```

Reply:
Done